app/smc/src/main.c

```diff
@@ -66,8 +70,15 @@ int main(void)
 		}
 	}
 
+	if (!DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(wdt0)) ||
+	    (get_fw_table()->chip_limits.auto_reset_timeout == 0)) {
```

Comment:
@palexsonTT - I guess there needs to be a `bool wdt0_disabled_by_fwtable = false`, and then inside of the `!...RECOVERY..` part, update it.

The condition at line 74 can be simplified to check the book variable.

```shell
/home/runner/work/tt-zephyr-platforms/tt-zephyr-platforms/zephyr-sdk/arc-zephyr-elf/bin/../lib/gcc/arc-zephyr-elf/12.2.0/../../../../arc-zephyr-elf/bin/ld.bfd: /home/runner/work/tt-zephyr-platforms/tt-zephyr-platforms/tt-zephyr-platforms/app/smc/src/main.c:74: undefined reference to `get_fw_table'
```